Tools/TestWebKitAPI/cocoa/WebExtensionUtilities.mm

```diff
@@ -355,8 +358,24 @@ - (void)_webExtensionController:(WKWebExtensionController *)controller receivedT
     [messagesArray addObject:argument ?: NSNull.null];
 }
 
+- (void)_webExtensionController:(WKWebExtensionController *)controller recordTestAddedWithTestName:(NSString *)testName andSourceURL:(NSString *)sourceURL lineNumber:(unsigned)lineNumber
+{
+    _testsAdded = [_testsAdded arrayByAddingObject:testName];
+}
+
+- (void)_webExtensionController:(WKWebExtensionController *)controller recordTestStartedWithTestName:(NSString *)testName andSourceURL:(NSString *)sourceURL lineNumber:(unsigned)lineNumber
+{
+    _runningTestFromQueue = YES;
+    _testsStarted = [_testsStarted arrayByAddingObject:testName];
+}
+
 - (void)_webExtensionController:(WKWebExtensionController *)controller recordTestFinishedWithResult:(BOOL)result message:(NSString *)message andSourceURL:(NSString *)sourceURL lineNumber:(unsigned)lineNumber
```

Comment:
This method likely should get a `testName` param, or have a new `testName` method in addition to this one. That way this can record the finish result per test by name too.

I think a dictionary like `testResults` would be good to update here when you have the test name.